### PR TITLE
Refactor CUDA bernoulli_kernel by using uniform_and_transform

### LINKED ISSUE #39557: Refactor CUDA bernoulli_kernel by using uniform_and_transform
`void bernoulli_kernel(TensorIterator& iter, double p_, RNG gen)` in `cuda/DistributionTemplates.h` can be simplified by using `uniform_and_transform` like it's used in other kernels in this file

Depends on #38558

### PR DESCRIPTION
**Pull request by Kiyosora**
- Fixes #39557 .
- Related #38558 .
- Simplifed `void bernoulli_kernel(TensorIterator& iter, double p_, RNG gen)` in `cuda/DistributionTemplates.h` by using `uniform_and_transform`.
- Unified `void bernoulli_kernel(TensorIterator& iter, double p_, RNG gen)` with other kernels in `cuda/DistributionTemplates.h`.

### CODE REVIEW

File: aten/src/ATen/core/DistributionsHelper.h [hunk 1]
```diff
@@ -186,21 +186,26 @@ struct normal_distribution {
     T stdv;
 };
 
+template <typename T>
+struct BernoulliType { using type = float; };
```
Comment (pbelevich) [inline aten/src/ATen/core/DistributionsHelper.h:190]: Let's combine `GeometricType` and `BernoulliType` to one `DiscreteDistributionType` and use it in `bernoulli_kernel` and `geometric_kernel`

**Review by pbelevich — CHANGES_REQUESTED** [review]
well done! just a couple of comments

File: aten/src/ATen/core/TransformationHelper.h [hunk 2]
```diff
@@ -131,4 +131,13 @@ C10_HOST_DEVICE inline T log_normal(T val) {
   return at::exp(val);
 }
 
+/**
+ * Transforms bernoulli distributed `val` between 0.0 and 1.0 to
```
[the remark this answers is not in the document]
Comment (Kiyosora) [reply aten/src/ATen/core/TransformationHelper.h]: Oops, Thanks for the correction!

File: aten/src/ATen/core/DistributionsHelper.h (hunk 1 above)
Comment (Kiyosora) [reply aten/src/ATen/core/DistributionsHelper.h]: addressed!

**Review by pbelevich — APPROVED** [review]

**Pull request closed without merging**